Source/DafnyStandardLibraries/src/Std/Parsers/String/StringParsers.dfy

```diff
@@ -0,0 +1,408 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT 
+ *******************************************************************************/
+
+/** Until string slicing is implemented in O(1), a string input consists of the original string,
+  * a start and an end marker. */
```

Comment:
:(

But consider extracting the `Input` datatype to a general `Collections.Seq.Slice<T>` datatype, since it's such a common workaround. I think the JSON library already has something similar too.